libs/server/Resp/RespServerSession.cs

```diff
@@ -210,13 +211,33 @@ public override int TryConsumeMessages(byte* reqBuffer, int bytesReceived)
                 clusterSession?.AcquireCurrentEpoch();
                 recvBufferPtr = reqBuffer;
                 networkSender.GetResponseObject();
-                ProcessMessages();
+
+                try
+                {
+                    ProcessMessages();
+                }
+                catch (RespParsingException ex)
+                {
+                    logger?.LogCritical($"Aborting open session due to RESP parsing error: {ex.Message}");
+                    logger?.LogDebug(ex, "RespParsingException in ProcessMessages:");
+
+                    // Forward parsing error as RESP error
+                    var resp = new ReadOnlySpan<byte>(Encoding.ASCII.GetBytes($"ERR Protocol Error: {ex.Message}"));
+                    while (!RespWriteUtils.WriteError(resp, ref dcurr, dend))
```

Comment:
nit: does the UTF16 -> ASCII encoding directly on the buffer
```suggestion
                    while (!RespWriteUtils.WriteError($"ERR Protocol Error: {ex.Message}", ref dcurr, dend))
```

Reply:
Resolved with the latest cleanup commit. Good catch as always! :)